fal/controllers/anime_stats.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+import configparser
+from typing import TYPE_CHECKING, Dict, Union, List, cast
+
+import jikanpy
+
+from fal.clients.mfalncfm_main import session_scope
+from fal.models import AnimeWeeklyStat, Season, Anime
+
+if TYPE_CHECKING:
+    from sqlalchemy.orm import Session
+
+config = configparser.ConfigParser()
+config.read("config.ini")
+
+
+def get_forum_posts(anime: Anime) -> int:
+    '''Not yet implemented'''
+    return 0
+
+
+def get_anime_stats_from_jikan(anime: Anime) -> Dict[str, Union[int, float]]:
+    '''
+    Makes requests to Jikan given an anime id
+    and returns a Dict containing all the information needed to
+    populate an AnimeWeeklyStat object
+    '''
+
+    jikan = jikanpy.Jikan()
+    general_anime_info = jikan.anime(anime.id)
+    jikan_anime_stats = jikan.anime(anime.id, extension='stats')
+
+    return {
+        'watching': jikan_anime_stats['watching'],
+        'completed': jikan_anime_stats['completed'],
+        'dropped': jikan_anime_stats['dropped'],
+        'score': general_anime_info['score'],
+        'favorites': general_anime_info['favorites'],
+        'forum_posts': get_forum_posts(anime)
+    }
+
+
+def populate_anime_weekly_stats() -> None:
+    '''
+    Populates the AnimeWeeklyStat table with a row for each anime
+    using data from Jikan.
+    '''
+
+    season_of_year = config.get("season info", "season").lower()
```

Comment:
Do we not need to annotate variables with type hints? I did in views/teams.py and jikanpy so I was wondering if it's unnecessary.

Reply:
I believe it's unnecessary when the type can be easily implied by the type checker (this would follow the same practice as statically typed languages like c++). In addition, in mypy you could turn up the strictness to ensure that all functions are annotated (which is what we've done) but there's no strictness level to enforce all variables annotated.

The only reference in the python community I could find was in this post though: https://realpython.com/python-type-checking/#variable-annotations

So I'm open to annotating all variables if there's a good reason to. It doesn't hurt though, so I never mentioned it in your PRs.

Reply:
Alright, that makes sense.